onnx/optimizer/optimize.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "onnx/py_utils.h" // Keep this at the top to avoid the compilation warning.
+
+#include <list>
+
+#include "onnx/ir.h"
+#include "onnx/ir_pb_converter.h"
+#include "onnx/optimizer/passes/eliminate_nop_transpose.h"
+#include "onnx/optimizer/passes/fuse_consecutive_transposes.h"
+#include "onnx/optimizer/passes/fuse_transpose_into_gemm.h"
+#include "onnx/optimizer/passes/split.h"
+
+namespace onnx { namespace optimization {
+
+void PrepareOutput(const onnx::ModelProto& mp_in, onnx::ModelProto& mp_out);
+
+struct Optimizer {
+  std::map<std::string, std::unique_ptr<OptimizePass>> passes;
+
+  Optimizer() {
+    // Register the optimization passes to the optimizer.
+    std::unique_ptr<FuseConsecutiveTransposes> fct(new FuseConsecutiveTransposes());
+    passes[fct->name] = std::move(fct);
+    std::unique_ptr<EliminateNopTranspose> ent(new EliminateNopTranspose());
+    passes[ent->name] = std::move(ent);
+    std::unique_ptr<FuseTransposeIntoGemm> ftg(new FuseTransposeIntoGemm());
+    passes[ftg->name] = std::move(ftg);
+    std::unique_ptr<SplitInit> si(new SplitInit());
+    passes[si->name] = std::move(si);
+    std::unique_ptr<SplitPredict> sp(new SplitPredict());
+    passes[sp->name] = std::move(sp);
+  }
+
+  std::string optimize(const std::string& content, std::list<std::string>& names) {
+
+    onnx::ModelProto mp_in;
+    ParseProtoFromBytes(&mp_in, content.c_str(), content.size());
+    std::shared_ptr<onnx::Graph> g = onnx::ImportModelProto(mp_in);
+    std::string out;
+
+    if (g.get() == nullptr) {
+      std::cerr << "Warning: onnx optimizer is unable to parse input model" << std::endl;
+      // If we can't parse the file, just return the original content.
+      out = content;
+    } else {
+      for (auto & name : names) {
+        auto it = passes.find(name);
+        if (it != passes.end()) {
```

Comment:
throw on unknown passes?

Reply:
Sure, we can do that. (warning is probably too weak. We should make sure that users know how to call optimizer correctly.)